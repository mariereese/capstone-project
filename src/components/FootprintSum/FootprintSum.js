import styled from 'styled-components/macro'
import Card from '../Card'
import FoodprintColorScale from '../FoodprintColorScale/FoodprintColorScale'

export default function FootprintSum({ sum, pointerPosition }) {
  return (
    <CardStyled>
      <h2>Foodprint:</h2>
      <p>
        {sum + ' '}kg CO<sub>2</sub>
      </p>
      <FoodprintColorScale pointerPosition={pointerPosition} />
    </CardStyled>
  )
}

const CardStyled = styled(Card)`
  h2 {
    margin-bottom: 0;
  }

  p {
    margin: 0;
    text-align: center;
    font-size: 2.25rem;
    font-weight: 400;
    color: var(--dark-grey);
  }

  sub {
    font-weight: 500;
  }
`
