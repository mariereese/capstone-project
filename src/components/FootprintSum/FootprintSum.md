```js
<FootprintSum sum={1.6} />
```
