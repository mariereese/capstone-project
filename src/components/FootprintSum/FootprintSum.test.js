import FootprintSum from './FootprintSum'
import { render } from '@testing-library/react'

describe('FoodprintSum', () => {
  it('renders correctly', () => {
    const { container } = render(<FootprintSum sum={0} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
