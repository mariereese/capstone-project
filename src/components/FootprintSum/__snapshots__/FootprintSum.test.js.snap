// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodprintSum renders correctly 1`] = `
<div
  class="Card-sc-1j6wwxg-0 FootprintSum__CardStyled-mr075j-0 gnpUlV eICXJW"
>
  <h2>
    Foodprint:
  </h2>
  <p>
    0 
    kg CO
    <sub>
      2
    </sub>
  </p>
  <div
    class="FoodprintColorScale__ColorScale-sc-1lwjm0x-0 kxLUjA"
  >
    <div
      class="FoodprintColorScale__Pointer-sc-1lwjm0x-1 cRBVUi"
    />
  </div>
</div>
`;
